infra/helper.py

```diff
@@ -523,36 +529,52 @@ def _get_latest_corpus(project_name, fuzz_target, base_corpus_dir):
 
 def download_corpus(project_name):
   """Download most recent corpus from GCS for the given project."""
+  try:
+    with open(os.devnull, 'w') as stdout:
+      subprocess.check_call(['gsutil', '--version'], stdout=stdout)
+  except OSError:
+    print('ERROR: gsutil not found. Please install it from '
+          'https://cloud.google.com/storage/docs/gsutil_install',
+          file=sys.stderr)
+    return False
+
   fuzz_targets = _get_fuzz_targets(project_name)
   corpus_dir = _get_corpus_dir(project_name)
   if not os.path.exists(corpus_dir):
     os.makedirs(corpus_dir)
 
   def _download_for_single_target(fuzz_target):
-    _get_latest_corpus(project_name, fuzz_target, corpus_dir)
+    try:
+      _get_latest_corpus(project_name, fuzz_target, corpus_dir)
+    except Exception as e:
+      print('ERROR: corpus download for %s failed: %s' % (fuzz_target, str(e)),
+            file=sys.stderr)
 
-  print('Downloading corpus for %s' % project_name)
+  print('Downloading corpus for %s project' % project_name)
   thread_pool = ThreadPool(multiprocessing.cpu_count())
   thread_pool.map(_download_for_single_target, fuzz_targets)
+  return True
```

Comment:
Thank you. All the issues I pointed out in https://github.com/google/oss-fuzz/pull/1494 are gone, but I came across a new one: `profile` runs fuzzers even if it has failed to download the corpus. For example, I used an account which didn't have access to the corpus of `systemd` at the time:
```sh
AccessDeniedException: 403 e...[at]gmail.com does not have storage.objects.list access to systemd-backup.clusterfuzz-external.appspot.com.
ERROR: corpus download for fuzz-dhcp-server failed: Command '['gsutil', 'ls', 'gs://systemd-backup.clusterfuzz-external.appspot.com/corpus/libFuzzer/systemd_fuzz-dhcp-server/']' returned non-zero exit status 1
AccessDeniedException: 403 e...[at]gmail.com does not have storage.objects.list access to systemd-backup.clusterfuzz-external.appspot.com.
ERROR: corpus download for fuzz-journal-remote failed: Command '['gsutil', 'ls', 'gs://systemd-backup.clusterfuzz-external.appspot.com/corpus/libFuzzer/systemd_fuzz-journal-remote/']' returned non-zero exit status 1
AccessDeniedException: 403 e...[at]gmail.com does not have storage.objects.list access to systemd-backup.clusterfuzz-external.appspot.com.
ERROR: corpus download for fuzz-unit-file failed: Command '['gsutil', 'ls', 'gs://systemd-backup.clusterfuzz-external.appspot.com/corpus/libFuzzer/systemd_fuzz-unit-file/']' returned non-zero exit status 1
AccessDeniedException: 403 e...[at]gmail.com does not have storage.objects.list access to systemd-backup.clusterfuzz-external.appspot.com.
ERROR: corpus download for fuzz-dns-packet failed: Command '['gsutil', 'ls', 'gs://systemd-backup.clusterfuzz-external.appspot.com/corpus/libFuzzer/systemd_fuzz-dns-packet/']' returned non-zero exit status 1
Running ./fuzz-dhcp-server
Running ./fuzz-journal-remote
Running ./fuzz-unit-file
Running ./fuzz-dns-packet
Serving the report on http://127.0.0.1:8008/
Serving HTTP on 0.0.0.0 port 8008 ...
```
It would probably be better to exit with an error code if something has gone wrong during downloading. Maybe something like
```diff
@@ -549,11 +549,13 @@ def download_corpus(project_name):
     except Exception as e:
       print('ERROR: corpus download for %s failed: %s' % (fuzz_target, str(e)),
             file=sys.stderr)
+      return False
+
+    return True

   print('Downloading corpus for %s project' % project_name)
   thread_pool = ThreadPool(multiprocessing.cpu_count())
-  thread_pool.map(_download_for_single_target, fuzz_targets)
-  return True
+  return not False in thread_pool.map(_download_for_single_target, fuzz_targets)
```
would work.

Reply:
Valid point! I'll address it on Monday, thanks!

Reply:
Done in https://github.com/google/oss-fuzz/commit/95a3d83a69a0fc9309d931c6b7c3dd4c2150295c, thanks for the patch suggestion!